src/Microsoft.Health.Dicom.Api/Controllers/WorkitemController.Retrieve.cs

```diff
@@ -18,14 +24,25 @@ public partial class WorkitemController
     /// This action requests a UPS Instance on the Origin-Server. It corresponds to the UPS DIMSE N-GET operation.
     /// </summary>
     [HttpGet]
+    [AcceptContentFilter(new[] { KnownContentTypes.ApplicationDicomJson })]
+    [Produces(KnownContentTypes.ApplicationDicomJson)]
+    [ProducesResponseType(typeof(DicomDataset), (int)HttpStatusCode.OK)]
+    [ProducesResponseType((int)HttpStatusCode.NoContent)]
     [VersionedPartitionRoute(KnownRoutes.RetrieveWorkitemInstancesRoute, Name = KnownRouteNames.VersionedPartitionRetrieveWorkitemInstance)]
     [PartitionRoute(KnownRoutes.RetrieveWorkitemInstancesRoute, Name = KnownRouteNames.PartitionedRetrieveWorkitemInstance)]
     [VersionedRoute(KnownRoutes.RetrieveWorkitemInstancesRoute, Name = KnownRouteNames.VersionedRetrieveWorkitemInstance)]
     [Route(KnownRoutes.RetrieveWorkitemInstancesRoute, Name = KnownRouteNames.RetrieveWorkitemInstance)]
     [AuditEventType(AuditEventSubType.RetrieveWorkitem)]
-    public IActionResult RetrieveAsync()
+    public async Task<IActionResult> RetrieveAsync(string workitemInstanceUid)
     {
-        _logger.LogInformation("Requesting non-implemented endpoint.");
-        return new StatusCodeResult((int)HttpStatusCode.NotFound);
+        _logger.LogInformation("Search workitem for WorkitemInstanceUid: '{WorkitemInstanceUid}'.", workitemInstanceUid);
+
+        var response = await _mediator
+            .RetrieveWorkitemAsync(workitemInstanceUid, cancellationToken: HttpContext.RequestAborted)
```

Comment:
Why HttpContext.RequestAborted?

Reply:
This is the cancellationtoken we are using for other operations as well.
HttpContext.RequestAborted - "Notifies when the connection underlying this request is aborted and thus request operations should be cancelled."